src/boot/efi/boot.c

```diff
@@ -770,7 +776,7 @@ static BOOLEAN menu_run(
                 if (firmware_setup) {
                         firmware_setup = FALSE;
                         if (key == KEYPRESS(0, 0, CHAR_CARRIAGE_RETURN))
-                                reboot_into_firmware();
+                                reboot_into_firmware(NULL, NULL, NULL);
```

Comment:
this is ugly. we used to do this a lot in our tree. but today I think it would be nicer if instead we'd define a unciton that does the actual work and then a wrapper around it that makes it something suitable as an entry callback.

i.e. leave reboot_into_firmware() the way it is, but add reboot_into_firmware_entry() as a trivial wrapper that takes the additional args and immediately calls reboot_into_firmware() and does nothing else.

The thing is that it oftentimes happens that people forget that we call this callbacks with invalid parameters too. And then they patch the callback in a way that assumes all params were valid and then in one corner case this will crash...